From 3b03f0f11ce3fff96a46eff90e125f8891b889a4 Mon Sep 17 00:00:00 2001
From: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Date: Tue, 25 Feb 2020 23:21:47 -0700
Subject: [PATCH 1244/1852] mtd: rawnand: pl353: Add new pattern to support
 write only

commit 59c2c4fda6812722806f82eb848db268438ca506 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds new pattern to support data writes only with wait ready
instruction. it is needed for BBT write.

Signed-off-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/nand/raw/pl353_nand.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/nand/raw/pl353_nand.c b/drivers/mtd/nand/raw/pl353_nand.c
index 1e5b05acdab3..4bb12a508fe1 100644
--- a/drivers/mtd/nand/raw/pl353_nand.c
+++ b/drivers/mtd/nand/raw/pl353_nand.c
@@ -913,6 +913,9 @@ static const struct nand_op_parser pl353_nfc_op_parser = NAND_OP_PARSER(
 		NAND_OP_PARSER_PAT_DATA_OUT_ELEM(true, PL353_MAX_CHUNK_SIZE),
 		NAND_OP_PARSER_PAT_CMD_ELEM(true),
 		NAND_OP_PARSER_PAT_WAITRDY_ELEM(true)),
+	NAND_OP_PARSER_PATTERN(pl353_nand_exec_op_cmd,
+		NAND_OP_PARSER_PAT_DATA_OUT_ELEM(false, PL353_MAX_CHUNK_SIZE),
+		NAND_OP_PARSER_PAT_WAITRDY_ELEM(true)),
 	);
 
 static int pl353_nfc_exec_op(struct nand_chip *chip,
-- 
2.31.1

